tests/e2e/specs/add-edit-category.test.js

```diff
@@ -0,0 +1,227 @@
+import { 
+	visitAdminPage,
+} from "@wordpress/e2e-test-utils";
+import { addQueryArgs } from "@wordpress/url";
+
+async function deleteAllCategories() {
+	const categoriesPageQuery = addQueryArgs( '', {
+		taxonomy: 'category',
+	} );
+	await visitAdminPage( 'edit-tags.php', categoriesPageQuery );
+
+	// Wait for the categories rows to appear
+	await page.waitForSelector( '#the-list tr' );
+
+	const allCategoriesRows = await page.$$( '#the-list tr' );
+	
+	// If there is more than one category row, delete all the categories
+	if( allCategoriesRows.length > 1 ) {
```

Comment:
I'm divided on whether we should check this. On one hand, there's a minor performance gain from not deleting/reloading the page if there's nothing to delete; on the other, it adds unnecessary complexity because we'll achieve exactly the same result whether we do this check or not.